coalib/parsing/FilterHelper.py

```diff
@@ -0,0 +1,38 @@
+from coalib.output.printers.LogPrinter import LogPrinter
+
+from coalib.parsing.filters.LanguageFilter import LanguageFilter
+from coalib.parsing.filters.CanDetectFilter import CanDetectFilter
+from coalib.parsing.filters.CanFixFilter import CanFixFilter
+
+
+class FilterHelper:
+    available_filters = {'language': LanguageFilter,
+                         'can_detect': CanDetectFilter,
+                         'can_fix': CanFixFilter}
+
+    @classmethod
+    def get_all_filters_str(cls, sep=', '):
+        return sep.join(sorted(cls.available_filters))
+
+    @classmethod
+    def is_valid_filter(cls, filter):
+        return filter in cls.available_filters
+
+    @classmethod
+    def get_filtered_bears(cls, filter, args, all_bears=None):
+        if all_bears is None:
+            from coalib.settings.ConfigurationGathering import (
+                get_all_bears)
+            all_bears = get_all_bears(LogPrinter())
+        if not args or len(args) == 0:
+            return all_bears
+        if cls.is_valid_filter(filter):
+            filter_object = cls.available_filters[filter]()
+            return filter_object.filter_bears(all_bears, args)
+
+        raise AttributeError(cls.get_help_message(filter))
+
+    @classmethod
+    def get_help_message(cls, filter):
+        return '{!r} is an invalid filter. Available filters: {}'.format(
+            filter, cls.get_all_filters_str())
```

Comment:
as you only use this message inside the error, I would rather inline it. Nobody would call `get_help_message` again ;)